osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/mbus/CoupleMBusDeviceCommandExecutor.java

```diff
@@ -42,6 +45,8 @@ public MbusChannelElementsResponseDto execute(final DlmsConnectionManager conn,
 
         LOGGER.debug("retrieving mbus info on e-meter");
 
+        deviceContext = new CommandExecutorDeviceContext(device);
```

Comment:
you are right, I did not check the caller logic and assumed each executor has it's own scope, because the Tools project acts based on new Executor().